src/NATS.Client.Core/INatsSerialize.cs

```diff
@@ -133,7 +133,19 @@ public void Serialize(IBufferWriter<byte> bufferWriter, T value)
         {
             if (value is DateTimeOffset input)
             {
-                if (Utf8Formatter.TryFormat(input, span, out var written))
+                bool result;
+                int written;
+                if (input.Offset == TimeSpan.Zero)
+                {
+                    // This will make it place `Z` instead of `+00:00` at the end
+                    result = Utf8Formatter.TryFormat(input.UtcDateTime, span, out written, new StandardFormat('O'));
```

Comment:
Serialization is a hot path, so it's worth noting one thing here...

[SharpLab seems to show better ASM for a `readonly` prop being referenced vs `new` even for simple case](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCEBXAMy5ilwDcNek1acefXCwAqMBBiG0AzE1IMAwgwDeNBnqYqAFMAgQANmgYBLAHYYAlAwCyhgMoAHbDYA8wAJ4YMAB8DAAmMLgYlgAi2IHSVvgwAPI8uDAYDABu2GbkjjrU+sXZ2FAMuAwAvAwAqhhcABwAYtD4cYFQMlB+rVDtGIY5eWjhkWgQHJm2mcAA7pY2MHMMrhheoWWhfQOGAOTJe/b2iiX6xADsDIa4aPMnuvoAvo96rwzuUFY5gQyw2KEIDYzH5VusbJsoNs2nEGAB9CDVBhLFZrDZbHZxfaHB5Fc5GEzmSwzRxOUhuTw+fyBEJjKIMWLxRIpNIZUp5ArvYo5cqVGr1JqYjCdbq9GGDYbkUYRKKTaZ2BjzSwI3FnPSXa63e6nZ40J5AA=)

[Looks like Ctor can throw which may prevent some inlining](https://github.com/dotnet/runtime/blob/edaa25e6942f3e8d9e2f078d99f34ed015658fbe/src/libraries/System.Private.CoreLib/src/System/Buffers/StandardFormat.cs#L56-L65) **However** for all of this pedanticism, it may or may not get handled by PGO when it's actually run. Might be worth trying to run a microbench between the two, while also noting 'real' implementation of a `readonly` based on gist could be `static` (alas, Sharplab  can't do statics easily) and likely could provide additional optimizations/benefits.